src/storage/DataAccessorBasedStore.ts

```diff
@@ -196,10 +196,11 @@ export class DataAccessorBasedStore implements ResourceStore {
       throw new ConflictHttpError(`${identifier.path} conflicts with existing path ${oldMetadata.identifier.value}`);
     }
 
-    const isContainer = this.isNewContainer(representation.metadata, identifier.path);
+    const isContainer = isContainerIdentifier(identifier);
     // Solid, §3.1: "Paths ending with a slash denote a container resource."
     // https://solid.github.io/specification/protocol#uri-slash-semantics
-    if (isContainer !== isContainerIdentifier(identifier)) {
+    const potentialSlug = representation.metadata.get(SOLID_HTTP.slug)?.value;
+    if (potentialSlug && (isContainerPath(potentialSlug) !== isContainerIdentifier(identifier))) {
```

Comment:
```suggestion
    if (isContainerType(representation.metadata) !== isContainerIdentifier(identifier)) {
```

You're actually right that this is something that was checked here due to the `isNewContainer` call, but that was just a consequence of that check having been added to that function later. That's on me for saying we needed the exact same behaviour 😅. But the PUT calls don't care about the slug so this can be removed.

The type check I suggest above is not strictly necessary either as we set the types correctly afterwards, but is probably still a good idea to prevent users from making mistakes.